compiler/rustc_mir/src/monomorphize/collector.rs

```diff
@@ -388,8 +388,27 @@ fn collect_items_rec<'tcx>(
                 collect_neighbours(tcx, instance, &mut neighbors);
             });
         }
-        MonoItem::GlobalAsm(..) => {
+        MonoItem::GlobalAsm(item_id) => {
             recursion_depth_reset = None;
+
+            let item = tcx.hir().item(item_id);
+            if let hir::ItemKind::GlobalAsm(asm) = item.kind {
+                for (op, op_sp) in asm.operands {
+                    match op {
+                        hir::InlineAsmOperand::Const { ref anon_const } => {
+                            // Treat these the same way as ItemKind::Const
+                            let anon_const_def_id =
+                                tcx.hir().local_def_id(anon_const.hir_id).to_def_id();
+                            if let Ok(val) = tcx.const_eval_poly(anon_const_def_id) {
+                                collect_const_value(tcx, val, &mut neighbors);
```

Comment:
The constant may refer to the address of a static in which case the static needs to be codegened as separate mono item.

Reply:
In general, yes. But not for the purposes of `global_asm!`, as far as I can tell. With this implementation we must be able to produce a string representation of the constant number of float within Rust already and Rust has no insight as to what the address of a static might look like.

Reply:
I've removed the `collect_const_value` but left most of the code around it since it will be needed to handle `sym` operands in the future (which *do* depend on other items).